src/blockchain/contracts/TwoPhaseCommit.sol

```diff
@@ -2,10 +2,76 @@
 pragma solidity >=0.4.22 <0.9.0;
 
 contract TwoPhaseCommit {
-  constructor() public {
-  }
+    enum Ballot {
+        UNKNOWN,
+        COMMIT,
+        ABORT
+    }
 
-  function getHeartBeat() public pure returns(bool) {
-      return true;
-  }
+    struct TransactionConfig {
+        uint256 cohorts;
+        uint256 vote_timeout_time;
+    }
+
+    // Voting states of each transaction.
+    // Mapping: transaction_id -> cohort_id -> ballot.
+    mapping(string => mapping(uint256 => Ballot)) private transaction_states;
+
+    // Configurations of each transaction.
+    mapping(string => TransactionConfig) private transaction_configs;
+
+    // Starts voting on a new transaction.
+    function startVoting(
+        string memory transaction_id,
+        // Number of cohorts which will be identified with cohort_id = 0, 1,
+        // ..., (cohorts -1).
+        uint256 cohorts,
+        // Format: unix timestamp.
+        uint256 vote_timeout_time
+    ) public {
+        transaction_configs[transaction_id].cohorts = cohorts;
+        transaction_configs[transaction_id]
+            .vote_timeout_time = vote_timeout_time;
+        for (uint256 i = 0; i < cohorts; i++) {
+            transaction_states[transaction_id][i] = Ballot.ABORT;
```

Comment:
As the Solidity unit-test doesn't fully support enum assertion, we are actually comparing int on enum type. Therefore, I saved UNKNOWN for unexpected unset value and we should avoid using that; otherwise, we might confuse unset value with a valid enum value.

I've updated the comment accordingly.